### PR TITLE
feat: Configure the lowest sequence to index for sequence-aware syncs

### PR DESCRIPTION
**Pull request by ameten**
### Description

When we use Sequence as indexing mode, now we can define the lowest sequence we would like to index. Before this change, the lowest sequence was always zero.

It will be the same lowest sequence for all the entities (dispatches, deliveries and payments). We will re-use field `IndexSettings.from` as the lowest sequence. Currently, the field is used only for watermark-aware indexing.

E2E Sealevel configuration was change so that we index sequence 0 since we start placing transaction from the very beginning.

### Backward compatibility

Yes

### Testing

Unit tests, E2E tests, Manual

### CODE REVIEW

**Review by yjamin — APPROVED** [review]

File: rust/main/hyperlane-base/src/contract_sync/cursors/sequence_aware/backward.rs [hunk 1]
```diff
@@ -108,37 +112,55 @@ impl<T: Debug + Clone + Sync + Send + Indexable + 'static> BackwardSequenceAware
...
+                not indexing anything below the lowest sequence"
+            );
+            return None;
+        }
+        let low = current_indexing_snapshot
             .sequence
-            .saturating_sub(self.chunk_size)..=current_indexing_snapshot.sequence
+            .saturating_sub(self.chunk_size)
+            // Use the latest sequence as the upper bound of the range.
+            .max(self.lowest_sequence);
```
[the remark this answers is not in the document]
Comment (ameten) [reply rust/main/hyperlane-base/src/contract_sync/cursors/sequence_aware/backward.rs:162]: I think that the comment is misleading. I wanted to say that the whole range should lay above the lowest sequence. I'll rephrase it.

File: rust/main/hyperlane-base/src/contract_sync/cursors/sequence_aware/backward.rs [hunk 2]
```diff
@@ -108,37 +112,55 @@ impl<T: Debug + Clone + Sync + Send + Indexable + 'static> BackwardSequenceAware
...
         current_indexing_snapshot: &TargetSnapshot,
-    ) -> RangeInclusive<u32> {
+    ) -> Option<RangeInclusive<u32>> {
         // Query the sequence range ending at the current_indexing_snapshot's sequence.
-        current_indexing_snapshot
+        // We assume that chunk size is at least 1 so that the sequence 0 is indexed
+        // together with sequence 1. That's why we can compare the current sequence
+        // with the lowest sequence with <=.
+        if current_indexing_snapshot.sequence <= self.lowest_sequence {
+            // If the current indexing snapshot's sequence is less than or equal to the lowest sequence,
```
[the remark this answers is not in the document]
Comment (yjamin) [reply rust/main/hyperlane-base/src/contract_sync/cursors/sequence_aware/backward.rs:148]: Okay in that case I would expect the 0th sequence to always be indexed even if there is no further one. Maybe you can add a test for this edge case?

File: rust/main/hyperlane-base/src/contract_sync/cursors/sequence_aware/backward.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ameten) [reply rust/main/hyperlane-base/src/contract_sync/cursors/sequence_aware/backward.rs:163]:
```
// Use the lowest sequence as the low bound of the range
// if the calculated low is less than the lowest sequence.
```

File: rust/main/hyperlane-base/src/contract_sync/cursors/sequence_aware/backward.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (ameten) [reply rust/main/hyperlane-base/src/contract_sync/cursors/sequence_aware/backward.rs:148]: It is a backward sequence-aware cursor. It starts from the highest sequence and makes its way to 0th sequence (or lowest configured sequence with this change).

If there is only 0th sequence (and no other sequences committed), this code won't index 0th sequence. I think we agree on this.

But if there is 0th sequence and 1th sequence and chunk size is 1, we'll index 0th sequence as well.

Since we always can produce more sequences, by dispatching more messages, I think it is ok not to index 0th sequence when there is only 0th sequence.
